### PR TITLE
Add com.neoutils.NeoRegex

### PR DESCRIPTION
**Pull request by Irineu333**
### Please confirm your submission meets all the criteria

- [x] Please describe the application briefly.
  A regular expression validator/tester.

- [x] The domain used for the application ID is [controlled by the application developers][appid-domain] and the [application id guidelines][appid] are followed.
- [x] I have read the and followed all the [Submission and licence requirements][reqs].
- [x] I have [built][build] and tested the submission locally.
- [x] I am an author/developer/upstream contributor of the project. If not, I contacted upstream developers about this submission. **Link:**

[appid-domain]: https://docs.flathub.org/docs/for-app-authors/requirements/#control-over-domain-or-repository
[appid]: https://docs.flathub.org/docs/for-app-authors/requirements#application-id
[reqs]: https://docs.flathub.org/docs/for-app-authors/requirements
[build]: https://docs.flathub.org/docs/for-app-authors/submission/#before-submission

### CODE REVIEW

**Comment by Irineu333** [comment]
I own the domain neoutils.com, how do I prove it?

File: com.neoutils.NeoRegex.yml [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+sdk: org.freedesktop.Sdk
+sdk-extensions:
+  - org.freedesktop.Sdk.Extension.openjdk17
+command: NeoRegex
+
+finish-args:
+  - --socket=x11
+  - --device=dri
+
+environment:
```
Comment (hfiguiere) [inline com.neoutils.NeoRegex.yml:13]: which page in the document say this is valid?

(so I can get it fixed)

Comment (Irineu333) [reply com.neoutils.NeoRegex.yml:13]: It wasn't based on the documentation. I confess that I didn't even consider the possibility of it being invalid. I apologize, I will correct it.

Comment (Irineu333) [reply com.neoutils.NeoRegex.yml]: It wasn't even necessary anymore. It ended up remaining as leftover from the initial tests.

File: com.neoutils.NeoRegex.yml [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+modules:
+  - name: openjdk
+    buildsystem: simple
+    build-commands:
+      - /usr/lib/sdk/openjdk17/install.sh
+
+  - name: NeoRegex
+    buildsystem: simple
+    build-commands:
+      - ls files
```
Comment (bbhtt) [inline com.neoutils.NeoRegex.yml:22]:
```suggestion
```

File: com.neoutils.NeoRegex.yml [hunk 3]
```diff
@@ -0,0 +1,33 @@
+app-id: com.neoutils.NeoRegex
```
Comment (bbhtt) [inline com.neoutils.NeoRegex.yml:1]: can you set up a link to https://github.com/NeoUtils/NeoRegex/ on https://www.neoutils.com/ ?

File: com.neoutils.NeoRegex.yml [hunk 4]
```diff
@@ -0,0 +1,33 @@
...
+      - ls files
+      - install -Dm755 files/bin/NeoRegex /app/bin/NeoRegex
+      - mkdir -p /app/lib
+      - cp -r files/lib/* /app/lib/
+      - install -Dm644 files/lib/NeoRegex.png /app/share/icons/hicolor/192x192/apps/com.neoutils.NeoRegex.png
+      - install -Dm644 files/com.neoutils.NeoRegex.desktop /app/share/applications/com.neoutils.NeoRegex.desktop
+      - install -Dm644 files/com.neoutils.NeoRegex.metainfo.xml /app/share/metainfo/com.neoutils.NeoRegex.metainfo.xml
+    sources:
+      - type: archive
+        url: https://github.com/NeoUtils/NeoRegex/releases/download/v2.0.0/NeoRegex-2.0.0.tar.gz
```
Comment (bbhtt) [inline com.neoutils.NeoRegex.yml:31]: This is x86_64 only:

```
file libapplauncher.so 
libapplauncher.so: ELF 64-bit LSB shared object, x86-64, version 1 (SYSV), dynamically linked, BuildID[sha1]=7b0fbf9e992648c4c9b7d4dea692450d66b14b04, with debug_info, not stripped
```

Comment (bbhtt) [reply com.neoutils.NeoRegex.yml:31]: https://docs.flathub.org/docs/for-app-authors/requirements#flathubjson

File: com.neoutils.NeoRegex.yml (hunk 3 above)
Comment (Irineu333) [reply com.neoutils.NeoRegex.yml:1]: I can yes

Comment (Irineu333) [reply com.neoutils.NeoRegex.yml:1]: Done

File: com.neoutils.NeoRegex.yml (hunk 4 above)
Comment (Irineu333) [reply com.neoutils.NeoRegex.yml:30]: Done.

**Comment by bbhtt** [comment]
bot, build com.neoutils.NeoRegex

**Comment by Irineu333** [comment]
I'm correcting it.

File: com.neoutils.NeoRegex.yml [hunk 5]
```diff
@@ -0,0 +1,32 @@
+app-id: com.neoutils.NeoRegex
+runtime: org.freedesktop.Platform
+runtime-version: '23.08'
+sdk: org.freedesktop.Sdk
+sdk-extensions:
+  - org.freedesktop.Sdk.Extension.openjdk17
+command: NeoRegex
+
+finish-args:
+  - --socket=x11
```
Comment (bbhtt) [inline com.neoutils.NeoRegex.yml:10]:
```suggestion
  - --socket=x11
  - --share=ipc
```

Comment (Irineu333) [reply com.neoutils.NeoRegex.yml:10]: Thanks

**Comment by Irineu333** [comment]
I ran the lint. It passed.

**Comment by bbhtt** [comment]
bot, build com.neoutils.NeoRegex

**Comment by Irineu333** [comment]
There was a problem with the icon. I'll need to give it some attention later, and when I fix it, I'll request a review again.

File: com.neoutils.NeoRegex.yml [hunk 6]
```diff
@@ -0,0 +1,33 @@
...
+    build-commands:
+      - /usr/lib/sdk/openjdk17/install.sh
+
+  - name: NeoRegex
+    buildsystem: simple
+    build-commands:
+      - install -Dm755 files/bin/NeoRegex /app/bin/NeoRegex
+      - mkdir -p /app/lib
+      - cp -r files/lib/* /app/lib/
+      - install -Dm644 files/lib/NeoRegex.png /app/share/icons/hicolor/192x192/apps/com.neoutils.NeoRegex.png
```
Comment (bbhtt) [inline com.neoutils.NeoRegex.yml:26]: this should be 128px or 256 px

Comment (Irineu333) [reply com.neoutils.NeoRegex.yml]: Thanks

**Comment by bbhtt** [comment]
bot, build com.neoutils.NeoRegex

File: com.neoutils.NeoRegex.yml (hunk 4 above)
Comment (hfiguiere) [reply com.neoutils.NeoRegex.yml]: why not building from source instead and have aarch64 support?

**Pull request closed without merging**